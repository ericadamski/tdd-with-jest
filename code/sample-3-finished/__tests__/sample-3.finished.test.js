const add = require("../sample-3.finished");

describe("Add", () => {
  it("should add two numbers", () => {
    // Arrange

    // Act
    const result = add(1, 2);

    // Assert
    expect(result).toBe(3);
  });

  it("should throw an error when attempting to add non-numbers", () => {
    // Arrange

    // Act
    // This is wrapped in a function
    // so that it doesn't crash when an error is thrown
    const boolResult = () => add(true, false);
    const stringResult = () => add("hello", "world");
    const arrayResult = () => add([0], [1, 2]);
    const objectResult = () => add({ value: 1 }, 1);
    const undefinedResult = () => add(undefined, 1);
    const nullResult = () => add(null, 1);
    const functionResult = () =>
      add(function() {
        return 1;
      }, 1);

    // Assert
    expect(boolResult).toThrowErrorMatchingSnapshot();
    expect(stringResult).toThrowErrorMatchingSnapshot();
    expect(arrayResult).toThrowErrorMatchingSnapshot();
    expect(objectResult).toThrowErrorMatchingSnapshot();
    expect(undefinedResult).toThrowErrorMatchingSnapshot();
    expect(nullResult).toThrowErrorMatchingSnapshot();
    expect(functionResult).toThrowErrorMatchingSnapshot();
  });
});
