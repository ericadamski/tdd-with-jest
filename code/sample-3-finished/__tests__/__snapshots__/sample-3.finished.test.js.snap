// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add should throw an error when attempting to add non-numbers 1`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 2`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 3`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 4`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 5`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 6`] = `"This isn't a number!"`;

exports[`Add should throw an error when attempting to add non-numbers 7`] = `"This isn't a number!"`;
